cell.go

```diff
@@ -6,8 +6,8 @@ package sqlite3
 
 // cellInfo holds information about an on-disk cell.
 type cellInfo struct {
-	Key          int64
-	RowID        int64
-	Payload      []byte
-	OverflowPage int32
+	LeftChildPage int32
+	RowID         *int64
```

Comment:
why the change to a `*int64` ?
could you explain this a bit?

Reply:
Not all btrees have rowids, so it's optional. But I think zero is a valid rowid. Actually, I'll check that...

Reply:
... oh yeah, unique integer columns alias to rowid, so zero is definitely allowed.

So you need a nil pointer to signal no rowid

Reply:
what about using -1 instead ?

Reply:
> The rowid of the SQL table is the 64-bit signed integer key for each entry in the table b-tree.

-1 is a valid value for RowIDs. There's no available, invalid value to signal a missing RowID.

Reply:
ok. I tried to reduce allocs :)